### PR TITLE
SOLR-16397: Create v2(JAX-RS) endpoints for RELOAD core

### PR DESCRIPTION
**Pull request by iamsanjay**
https://issues.apache.org/jira/browse/SOLR-16397

# Description

V2(JAX-RS) Endpoints for core admin op "RELOAD".

# Solution

There is already one class present in the codebase named as ReloadCoreAPI.java which is using home-grown V2 api framework, this New code code changes will migrate it to JAX-RS framework.

# Tests

Added two test cases:

1. To test the valid response and status message.
2. Test Non Existent core exception and status code.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [x] I have added tests for my changes.
- [x] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Review by gerlowskija — APPROVED** [review]
Looks great, thanks Sanjay!  No real notes, other than a typo in a Javadoc comment.

Will aim to test and merge this later in the week!

**Comment by iamsanjay** [comment]
@gerlowskija I have tried the POST request without `Content-type` header,  and it returns `415 Unsupported Media Type`. Why I am saying this? Because as per our last discussion, It supposed to work without this header and hence, we removed it from documentation.

File: solr/core/src/java/org/apache/solr/handler/admin/api/ReloadCoreAPI.java [hunk 1]
```diff
@@ -17,52 +17,69 @@
...
-    final Map<String, Object> v1Params = new HashMap<>();
-    v1Params.put(ACTION, CoreAdminParams.CoreAdminAction.RELOAD.name().toLowerCase(Locale.ROOT));
-    v1Params.put(CoreAdminParams.CORE, coreName);
-
-    coreHandler.handleRequestBody(wrapParams(obj.getRequest(), v1Params), obj.getResponse());
+  @POST
+  @Produces({"application/json", "application/xml", BINARY_CONTENT_TYPE_V2})
+  @PermissionName(CORE_EDIT_PERM)
+  public SolrJerseyResponse reloadCore(
+      @Parameter(description = "The name of the core to snapshot.", required = true)
```
Comment (gerlowskija) [inline solr/core/src/java/org/apache/solr/handler/admin/api/ReloadCoreAPI.java:62]: [-0] Gah, minor copy/paste error here it looks like.

File: solr/core/src/java/org/apache/solr/handler/admin/api/ReloadCoreAPI.java [hunk 2]
```diff
@@ -17,52 +17,69 @@
...
+          @PathParam("coreName")
+          String coreName,
+      @Schema(description = "The POJO for representing additional backup params.") @RequestBody
```
Comment (gerlowskija) [inline solr/core/src/java/org/apache/solr/handler/admin/api/ReloadCoreAPI.java:65]: [-0] Ditto, re: copy/paste error

**Comment by gerlowskija** [comment]
> I have tried the POST request without Content-type header, and it returns 415 Unsupported Media Type. Why I am saying this? Because as per our last discussion, It supposed to work without this header and hence, we removed it from documentation.

Do you have a pointer to that discussion, by chance?  I believe you if that's what I said, but maybe there was surrounding context that I should've made explicit or it was in some exceptional case?

In general, I think we should allow users to skip the `Content-type` header when there is no request-body.  But if a user is providing content, requiring `Content-type` seems reasonable IMO.  It's a pretty standard practice in HTTP APIs.  Maybe we could add some "best guess" attempt to try to determine the content type (I know I've seen code like this elsewhere in Solr, will have to look that up...).  But that can be handled independent of this ticket IMO

**Review by gerlowskija — COMMENTED** [review]
Noticed a few last minute nits as I was testing.  Will fix these myself, but still leaving the comments for tracking purposes.

**Comment by gerlowskija** [comment]
Thanks for the contribution Sanjay!  I've merged to `main` and will try to backport for 9.4 towards the end of this week!

**Comment by iamsanjay** [comment]
Thank you so much for the reviewing it and fixing few issues as well. Just one thing, I removed the 'Content-type' from the documentation.  I will make another pull request for documentation change and then please merge that as well.

**Pull request merged**